core/src/main/java/org/apache/calcite/rex/RexBuilder.java

```diff
@@ -1359,7 +1359,7 @@ public RexNode makeBetween(RexNode arg, RexNode lower, RexNode upper) {
   /** Converts a list of expressions to a search argument, or returns null if
    * not possible. */
   private static <C extends Comparable<C>> Sarg<C> toSarg(Class<C> clazz,
-      List<? extends RexNode> ranges, boolean containsNull) {
+      RelDataType type, List<? extends RexNode> ranges, boolean containsNull) {
```

Comment:
As you are modifying this signature, could you update the java doc to document what theses parameters are?

This will help understand why need a type.

Reply:
Sounds reasonable. Thanks. 
Here we need a type parameter, because we need it when calling `Sarg.of`.
So I have updated the Javadoc for `Sarg.of` to explain the reason for the new parameter.